libdrgn/dwarf_index.c

```diff
@@ -2608,6 +2637,9 @@ drgn_dwarf_index_update(struct drgn_dwarf_index_update_state *state)
 	if (err)
 		goto err;
 
+	if ((err = drgn_dwarf_index_namespace_shards_init(&dindex->global)))
+		return err;
```

Comment:
Can we do this as the first thing in `drgn_dwarf_index_update()`? Then we'll never need to roll back before we have any shards.